### DISCUSSION TITLE
MAUI for OpenGL Games! (e.g. Unity3D, Urho3D.NET, Stride, Wave, Godot, Open3D)

### DISCUSSION THREAD

**Discussion by najak3d** [dotnet/Microsoft.Maui.Graphics.Controls#93] [Ideas]
One of our primary objectives here is to enable MAUI to be rendered to a Bitmap rendered to an OpenGL surface as the HUD/UI. Each MAUI layout (panel, stack, window) - would be associated to it's own Bitmap, which is then handed off to OpenGL to render to screen like a simple fast-updating Picture box (with transparency and click-through). It's about time that Game Engines stop trying to re-invent their own custom UI's, and just allow everyone to use a common UI package, like MAUI. MAUI rendering to a Unity3D app, is one of the home runs we hope to achieve here.

Note: We've already accomplished this feat with Avalonia and Urho3D.NET. But realize that the Home-Run solution isn't Avalonia -- but instead is MAUI.Graphics.Controls.

Maui.Graphics.Controls -- is the key to this success, for Msft/MAUI.

We may be intending to help you complete the missing controls (e.g. List View, Tree View, Scroll View, ComboBox, ...).

Please - someone inside Msft, please shout at the Msft MAUI team, to see what it takes to put more priority onto this Graphics.Controls lib! It's a VERY big deal!

===
NOTE: As quickly as we can do it, our intent is to produce a POC of this "rendering MAUI inside an OpenGL surface", to help boost the energy/priority of this concept. We believe it should be a "front-page" feature of MAUI, with raised priority for development. In conjunction with this, our current aim is to also make use of Comet.

Headline: Maui.MVU rendering to Graphics.Controls Anywhere. (even Games, and web pages)

**Comment by IanBUK** [comment]
I'd be interested in the inverse too - building a Maui app that has a 3D frame for displaying a mesh.

**Reply by freever** [reply]
Definitely. Something like the MAUI GraphicsView but embedding a Veldrid window would be awesome

**Comment by Another5405** [comment]
I would love to know any update if have.

**Comment by akhanalcs** [comment]
This would be super cool!

**Comment by anose001** [comment]
Any update or development in this discussion?
